File: src/js/handlers/primitives.ts
import { registerHandler } from "../handler";

const handler: Handler = {
    priority: 10,
    from(obj: unknown): unknown {
        if (typeof obj === "boolean" || typeof obj === "number" || typeof obj === "string" || obj === null) {
            return obj;
        }
        return undefined;
    },
    to(obj: unknown): unknown {
        if (typeof obj === "bigint") {
            return obj.toString();
        }
        if (typeof obj === "boolean" || typeof obj === "number" || typeof obj === "string" || typeof obj === "undefined" || obj === null) {
            return obj;
        }
        return undefined;
    }
}

registerHandler(handler);

File: src/lua/handlers/primitives.lua
local HANDLER = {}

HANDLER.Priority = 10

function HANDLER:From(obj)
    -- Convert from primitive types to themselves
    local t = type(obj)
    if t == "nil" or t == "boolean" or t == "number" or t == "string" then
        return obj
    end
end

function HANDLER:To(obj)
    -- Convert from primitive types to themselves
    local t = type(obj)
    if t == "nil" or t == "boolean" or t == "number" or t == "string" then
        return obj
    end
end

Interop:RegisterHandler(HANDLER)